activesupport/lib/active_support/message_pack/extensions.rb

```diff
@@ -86,8 +86,9 @@ def install(registry)
           unpacker: URI.method(:parse)
 
         registry.register_type 14, IPAddr,
-          packer: :to_s,
-          unpacker: :new
+          packer: method(:write_ipaddr),
+          unpacker: method(:read_ipaddr),
+          recursive: true
```

Comment:
The problem is that we can't just change the serialization of a type now that the first version was released, as previous payload wouldn't be readable, and old code wouldn't be able to read new payload (cc @jonathanhefner that's the kind of stuff I was worried about :/)

So now we need to define a new `IPAddr` type, and conditionally enable it, [like we do in `Paquito`](https://github.com/Shopify/paquito/blob/main/lib/paquito/types.rb).

Reply:
Ah, you are right. We should consider backward compatibility.

Reply:
write the single string `"#{ipaddr.to_s}/#{ipaddr.prefix}"` as single packer write, and unpacker back to `:new`.
I think this way would keep the compatibility, but what do you think?

Reply:
okey, did it without deprecation system.

Reply:
@byroot As @r-plus mentioned, a new type isn't necessary in this case.

Reply:
@byroot As @r-plus mentioned, a new type isn't necessary in this case.

Reply:
I see it's not needed for backward compatibility, but what about forward compatibility? We unfortunately need both to allow for smooth upgrades.

Reply:
Nevermind, I see the `unpack` is unchanged, so we're good.